src/Nethermind/Nethermind.Shutter/ShutterBlockHandler.cs

```diff
@@ -0,0 +1,150 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Consensus.Processing;
+using Nethermind.Core;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Shutter.Contracts;
+using Nethermind.Logging;
+using Nethermind.Abi;
+using Nethermind.Blockchain.Receipts;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Collections.Concurrent;
+using Nethermind.Core.Caching;
+using Nethermind.Core.Crypto;
+using Nethermind.Blockchain;
+using Nethermind.Core.Collections;
+
+namespace Nethermind.Shutter;
+
+public class ShutterBlockHandler(
+    ulong chainId,
+    string validatorRegistryContractAddress,
+    ulong validatorRegistryMessageVersion,
+    ReadOnlyTxProcessingEnvFactory envFactory,
+    IReadOnlyBlockTree blockTree,
+    IAbiEncoder abiEncoder,
+    IReceiptFinder receiptFinder,
+    Dictionary<ulong, byte[]> validatorsInfo,
+    IShutterEon eon,
+    ShutterTxLoader txLoader,
+    ShutterTime shutterTime,
+    ILogManager logManager,
+    TimeSpan slotLength,
+    TimeSpan blockWaitCutoff) : IShutterBlockHandler
+{
+    private readonly ArrayPoolList<CancellationTokenSource> _cts = new(10);
+    private readonly ArrayPoolList<CancellationTokenRegistration> _ctr = new(5);
```

Comment:
If it lives for the whole duration of the application, no need for Pooling.
Also it seems you only add to this collection, never remove, does this leak memory?